dev/source/docs/sim-on-hardware.rst

```diff
@@ -0,0 +1,90 @@
+- ensuring that physical failsafes (e.g. parachutes) function appropriately without putting a vehicle at risk
+- communications infrastructure works in the face of real traffic
+
+.. note::
+
+   Simulation on Hardware is being improved all the time.  It is suggested you run the master branch ("latest" firmware) to be able to take advantage of any new features that have been added.
+
+The firmware is loaded on the autopilot and a GCS attached. The GPS initial location data is provided by the simulation and can be changed when compiling the firware. Setup, arming, and flying the simulation occurs as if sim_vehicle.py and the associated physics models for the vehicle and frame was used.
+
+Limitations
+===========
+
+-  Actuator outputs (motor and servo) are not supported in the firmware as posted on the :ref:`firmware server <https://firmware.ardupilot.org>` without recompilation.  This is currently considered a safety feature.
+
+.. warning:: relay outputs currently are not prevented from operation. Be careful if they are being used for payload releases or parachute deployment.
+
+-  Parameter space is shared between the real aircraft and the simulated aircraft.  This leads to problems with sensor calibration as the sensor suite differs between the real and virtual aircraft.  Thus parameters must be wiped and rewritten to the vehicle when moving between real and simulation firmware.
+- While it is possible to run on flash-constrained (1MB) boards, many other features will need to be compiled out to allow it to fit.
+- Only Quadcopters have been tested as working.
+- Currently only firmware for Cube Orange QuadCopter autopilots is available pre-compiled on the firmware server.
+
+
+Firmware
+========
+
+We provide a pre-compiled binary for Cube Orange at :ref:`our firmware server <https://firmware.ardupilot.org>`. It is available only in :ref:`Copter <https://firmware.ardupilot.org/Copter/latest/CubeOrange-SimOnHardWare/>` version, currently.
+
+Compiling your own Firmware
+============================
+
+Simulation on Hardware is not compiled into the ArduPilot firmware by default.
+
+If you have setup an :ref:`ArduPilot development environment <dev:building-the-code>` you can add it to any ChibiOS hwdef file by adding the line `env SIM_ENABLED 1`.  Use the CubeOrange-SimOnHardware hwdef file (and default parameter file!) as a reference.
+
+A typical build script is shown below:
+
+.. code:: bash
+
+    set -e
+    set -x
+
+    BOARD=MatekH743
+    THISDIR=$(dirname $0)
+
+    ./waf configure \
+      --board=$BOARD \
+      --extra-hwdef="$THISDIR/extra-hwdef-sitl-on-hw.dat" \  a typical file is shown `here <https://github.com/ArduPilot/ardupilot/blob/master/libraries/SITL/examples/on-hardware/extra-hwdef-sitl-on-hw.dat>`__
+      --default-param="$THISDIR/default.param" \ a typical default param file is shown `here <https://github.com/ArduPilot/ardupilot/blob/master/libraries/SITL/examples/on-hardware/default.param>`__
```

Comment:
Perhaps move the references to the typical files to the bottom to easy copy/paste?